dbptk-modules/dbptk-module-siard/src/main/java/com/databasepreservation/modules/siard/in/metadata/SIARDDK128MetadataImportStrategy.java

```diff
@@ -0,0 +1,472 @@
+    } catch (JAXBException e) {
+      throw new ModuleException().withMessage("Error loading JAXBContext").withCause(e);
+    }
+
+    SchemaFactory schemaFactory = SchemaFactory.newInstance(XMLConstants.W3C_XML_SCHEMA_NS_URI);
+    Schema xsdSchema = null;
+    InputStream xsdInputStream = new FileInputStream(pathStrategy.getMainFolder().getPath().toString() + "/"
+            + pathStrategy.getXsdFilePath(SIARDDKConstants.DOC_INDEX));
+
+    try {
+      xsdSchema = schemaFactory.newSchema(new StreamSource(xsdInputStream));
+    } catch (SAXException e) {
+      throw new ModuleException()
+              .withMessage("Error reading metadata XSD file: " + pathStrategy.getXsdFilePath(SIARDDKConstants.DOC_INDEX))
+              .withCause(e);
+    }
+    InputStream inputStreamXml = null;
+    Unmarshaller unmarshaller;
+    try {
+      unmarshaller = context.createUnmarshaller();
+      unmarshaller.setSchema(xsdSchema);
+      inputStreamXml = new FileInputStream(pathStrategy.getMainFolder().getPath().toString() + "/" +
+              pathStrategy.getXmlFilePath(SIARDDKConstants.DOC_INDEX));
+      JAXBElement<DocIndexType> jaxbElement = (JAXBElement<DocIndexType>) unmarshaller.unmarshal(inputStreamXml);
+      return jaxbElement.getValue();
+    } catch (JAXBException e) {
+      throw new ModuleException().withMessage("Error while Unmarshalling JAXB").withCause(e);
+    } finally {
+      try {
+        xsdInputStream.close();
+        if (inputStreamXml != null) {
+          inputStreamXml.close();
+          xsdInputStream.close();
+        }
+      } catch (IOException e) {
+        logger.debug("Could not close xsdStream", e);
+      }
+    }
+  }
+
+  private List<ForeignKey> getVirtualForeignKeys(ColumnsType columns, String tableId) {
+    List<ForeignKey> virtualForeignKeys = new ArrayList<>();
+    for (ColumnType column : columns.getColumn()) {
+      if (column.getFunctionalDescription() != null && column.getFunctionalDescription().contains(FunctionalDescriptionType.DOKUMENTIDENTIFIKATION)) {
+        ForeignKey virtualForeignKey = new ForeignKey();
+        virtualForeignKey.setReferencedSchema(getImportAsSchemaName());
+        virtualForeignKey.setName("FK_virtual_table");
+        virtualForeignKey.setReferencedTable("virtual_table");
+        Reference reference = new Reference(column.getName(), "dID");
+        List<Reference> referenceList = new ArrayList<>();
+        referenceList.add(reference);
+        virtualForeignKey.setReferences(referenceList);
+        virtualForeignKey.setId(String.format("%s.%s", tableId, "FK_virtual_table"));
+        virtualForeignKeys.add(virtualForeignKey);
+      }
+    }
+
+    return virtualForeignKeys;
+  }
+
+  private TableStructure createVirtualTable() throws ModuleException {
+    try {
+      DocIndexType docIndexType = loadVirtualTableMetadata();
+      TableStructure virtualTable = new TableStructure();
```

Comment:
Create a new VirtualTableStructure class that extends TableStructure, in the constructor of this class configure the object with everything that a virtual table needs to have by default, such as name, description, etc.

With this you will be able to differentiate a table from a virtual table using intanceof, which can be useful for applications that use dbptk-developer as a library.

I will review the rest of the class after this change

Reply:
done